refactor(serviceaccount-tokens-controller): Change MutationCache to SecretLister

zk-123 opened the pull request:
Change MutationCache to SecretLister



#### What type of PR is this?

/kind cleanup
/kind deprecation


#### What this PR does / why we need it:

In the tokens-controller, Mutation Secret is no longer needed. Using SecretLister makes the code structure clearer and easier to read.


#### Which issue(s) this PR is related to:
N/A

#### Special notes for your reviewer:
I believe I did not change the code logic, and change 

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

zk-123 committed:
refactor(serviceaccount-tokens-controller): Change the secret's MutationCache to SecretLister

zk-123 commented:
/easycla

liggitt commented on pkg/controller/serviceaccount/tokens_controller.go:
```diff
@@ -510,54 +507,31 @@ func (e *TokensController) getServiceAccount(ns string, name string, uid types.U
-	obj, exists, err := e.updatedSecrets.GetByKey(makeCacheKey(ns, name))
-	if err != nil {
+	secret, err := e.secrets.Secrets(ns).Get(name)
+	if err != nil && !apierrors.IsNotFound(err) {
```
return `nil, nil` early if `apierrors.IsNotFound(err)`:

```go
if err != nil {
  if apierrors.IsNotFound(err) {
    return nil, nil
  }
  return nil, err
}
```

liggitt commented on pkg/controller/serviceaccount/tokens_controller.go:
```diff
@@ -510,54 +507,31 @@ func (e *TokensController) getServiceAccount(ns string, name string, uid types.U
-	}
 	// Ensure UID matches if given
-	if len(uid) == 0 || uid == secret.UID {
+	if err == nil && (len(uid) == 0 || uid == secret.UID) {
```
revert this change, we'll return earlier if err is non-nil

liggitt commented on pkg/controller/serviceaccount/tokens_controller_test.go:
```diff
@@ -483,9 +483,6 @@ func TestTokenCreation(t *testing.T) {
 				secrets.Add(tc.AddedSecret)
 				controller.queueSecretSync(tc.AddedSecret)
 			}
-			if tc.AddedSecretLocal != nil {
```
this is the only use of AddedSecretLocal ... dropping this means a testcase using that is probably not testing what it thinks it is

zk-123 commented:
thank you very much for your guidance. 

- return earlier is a great suggestion, and I’ve already fixed it. 
- I’ve also removed the test case for "AddLocalSecretToCache" since it’s no longer needed.

@liggitt

pacoxu commented:
/remove-kind deprecation
`kind/deprecation` is generally used for KEP which is marked for deprecation.

pacoxu commented:
/release-note-none
